python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -514,13 +514,58 @@ class AssetBackfillIterationResult(NamedTuple):
     backfill_data: AssetBackfillData
 
 
+def _get_requested_asset_partitions_from_run_requests(
+    run_requests: Sequence[RunRequest],
+    asset_graph: ExternalAssetGraph,
+    instance_queryer: CachingInstanceQueryer,
+) -> Set[AssetKeyPartitionKey]:
+    requested_partitions = set()
+    for run_request in run_requests:
+        # Run request targets a range of partitions
+        range_start = run_request.tags.get(ASSET_PARTITION_RANGE_START_TAG)
+        range_end = run_request.tags.get(ASSET_PARTITION_RANGE_END_TAG)
+        if range_start and range_end:
+            # When a run request targets a range of partitions, each asset is expected to
+            # have the same partitions def
+            selected_assets = cast(Sequence[AssetKey], run_request.asset_selection)
+            check.invariant(len(selected_assets) > 0)
+            partitions_defs = set(
+                asset_graph.get_partitions_def(asset_key) for asset_key in selected_assets
+            )
+            check.invariant(
+                len(partitions_defs) == 1,
+                "Expected all assets selected in partition range run request to have the same"
+                " partitions def",
+            )
+
+            partitions_def = cast(PartitionsDefinition, next(iter(partitions_defs)))
+            partitions_in_range = partitions_def.get_partition_keys_in_range(
+                PartitionKeyRange(range_start, range_end), instance_queryer
+            )
+            requested_partitions = requested_partitions | set(
```

Comment:
You can use a [set comprehension](https://www.google.com/search?q=set+comprehension+python&rlz=1C5CHFA_enUS729US729&oq=set+comprehe&aqs=chrome.0.0i512j69i57j0i512l7j0i390i650.1873j0j7&sourceid=chrome&ie=UTF-8) instead of passing a list to `set`. Same with below.